Automattic/wp-calypso
# reduxify `Security2faSMSSettings` and drop `formBase`

#### Changes proposed in this Pull Request

* Reduxify `Security2faSMSSettings`
* Drop the `formBase` mixin and refactor to class component

#### Testing instructions

* Get a WP.com account which doesn't have two-factor authentication enabled
* Go to `/me/security/two-step`
* Go through the setup flow and make sure it's behaving exactly like on prod
* At the first step, try blocking the network request before you hit on _Verify by X_ to simulate an error case for saving the phone number

part of #24162
